File: java/src/com/nttdocomo/io/HttpConnection.java
package com.nttdocomo.io;


import java.io.DataInputStream;
import java.io.InputStream;
import java.io.OutputStream;

public interface HttpConnection {
    InputStream openInputStream();

    OutputStream openOutputStream();

    void close();

    DataInputStream openDataInputStream();

    void setRequestMethod(String get);

    void connect();

    long getLength();
}

File: web/src/com/nttdocomo/io/HttpConnection.ts
import type { DataInputStream } from '../../../java/io/DataInputStream';
import type { InputStream } from '../../../java/io/InputStream';
import type { OutputStream } from '../../../java/io/OutputStream';

export interface HttpConnection {
	setRequestMethod(method: string): void;
	connect(): Promise<void>;
	openInputStream(): Promise<InputStream>;
	openDataInputStream(): Promise<DataInputStream>;
	openOutputStream(): Promise<OutputStream>;
	getLength(): Promise<number>;
	close(): Promise<void>;
}
